lib/codecs/tests/gelf.rs

```diff
@@ -0,0 +1,112 @@
+use bytes::Bytes;
+use chrono::{DateTime, NaiveDateTime, Utc};
+use codecs::{
+    decoding::format::gelf_fields::*, decoding::format::Deserializer, GelfDeserializerConfig,
+};
+use pretty_assertions::assert_eq;
+use serde_json::json;
+use value::Value;
+use vector_core::config::log_schema;
+
+/// Validates all the spec'd fields of GELF are deserialized.
+#[test]
+fn gelf_deserializing_all() {
+    let config = GelfDeserializerConfig;
+    let deserializer = config.build();
+
+    let add_on_int_in = "_an.add-field_int";
+    let add_on_str_in = "_an.add-field_str";
+    let add_on_int_out = "_an_add_field_int";
+    let add_on_str_out = "_an_add_field_str";
+
+    let input = json!({
+        VERSION: "1.1",
```

Comment:
My reading of this is that GELF is case-insensitive. Is that right?

Reply:
Aha, I missed that these were constants. I was thinking the `json!` macro just allowed unquoted text for keys. 👍

Reply:
Ignore this, I missed that `VERSION` and others were constants.

Reply:
Doh, commented before I saw your new comment. I think the missing piece was that I didn't realize these were constants.